crates/factor-outbound-http/src/wasi.rs

```diff
@@ -61,10 +70,261 @@ impl<'a> WasiHttpView for WasiHttpImplInner<'a> {
 
     fn send_request(
         &mut self,
-        _request: Request<wasmtime_wasi_http::body::HyperOutgoingBody>,
-        _config: wasmtime_wasi_http::types::OutgoingRequestConfig,
+        mut request: Request<wasmtime_wasi_http::body::HyperOutgoingBody>,
+        mut config: wasmtime_wasi_http::types::OutgoingRequestConfig,
     ) -> wasmtime_wasi_http::HttpResult<wasmtime_wasi_http::types::HostFutureIncomingResponse> {
-        // TODO: port implementation from spin-trigger-http
-        Err(ErrorCode::HttpRequestDenied.into())
+        if let Some(interceptor) = &self.state.request_interceptor {
+            let intercepted = Intercepted {
+                request: &mut request,
+                config: &mut config,
+            };
+            if let Some(res) = interceptor.intercept(intercepted) {
+                return res;
+            }
+        }
+
+        let host = request.uri().host().unwrap_or_default();
```

Comment:
Are we asking for issues if `host` is `""`? I guess this _should_ return the default tls config, and everything will just work, but it does seem a bit round about, and I worry things might get confusing with future code changes.

Reply:
Well, `host` could also be present-but-empty I think? Or do your recent changes make that impossible?

Reply:
I'm actually not really sure. I *think* from a logic point of view this is probably fine. Then again, just because something works doesn't mean it's the right implementation. I personally would likely feel better with a more verbose and explicit implementation, but I won't die on this hill 😄.